run_tests.sh

```diff
@@ -0,0 +1,7 @@
st,RDFDatetimeTest,FilehashTest,XoredSearchingTest,GRRFuseTest,ClientCommsTest,FilterTests,AFF4SparseImageTest,ByteSizeTest,OSXEnumerateRunningServicesTest,TimelineTest,TestTimelines,TestPackedVersionedCollection,VFSDirectoryTest,FindTest,RekallTests,RDFIntegerTest,TestExecutePython,MethodTest,EmailPluginTest,PackedVersionedCollectionCompactorTest,OSXMemoryTest,HashFileStoreTest,ArtifactFlowTest,AFF4ObjectLabelsListTest,TestExports,AuthTest,ConfigLibTest,GrepTests,GrrKbTest,TestArtifactCollectors,TriggersTest,UsersTest,SshdConfigTest,TestLinuxMemory,DurationTest,ClientURNTests,SqliteDataStoreTest,DataStoreTest,AFF4ObjectLabelTest,TestCheckFlows,TDBDataStoreBenchmarks,ThreadPoolTest,AFF4GRRTest,ChromeHistoryTest,GetClientStatsActionTest,TestSearchFileContent,TestRegistryFinderFlow,AFF4LabelsIndexTest,ListProcessesTest,Ipv6UtilsTest,RDFStructsTest,BuildTests,TestPlistFlows,FileStoreHashTest,ApiConfigRendererTest,IEHistoryTest,RunningStatsTest,AdministrativeFlowTests,YamlConfigTest,RDFValueBaseTest,RDFBytesTest,FileExportPluginTest,TestCopyPathToFile,SqliteDataStoreCSVBenchmarks,RDFValueBenchmark,TDBDataStoreTest,TestLinuxMemoryAnalysis,StoreTests,DataStoreBenchmarks,HintsTests,RekallResponseConverterTest,WindowsRegistryParserTest,ClientUtilsTest,GRRTempFileTestDirectory,EmptyActionTest,FakeDataStoreBenchmarks,AverageMicroBenchmarks,TestListVolumeShadowCopies,WindowsPersistenceMechanismsParserTest,HashDigestTest,WMIParserTest,CollectionExportPluginTest,StatsTests,MasterTest,ApiAff4RendererTest,SendEmailTests,TestClientInterrogate,TestExportHuntResultsFilesAsArchive,RDFURNTest,LinuxSoftwareParserTest,ArtifactParserTest,TestFilesystem,HuntTest,BatchConverterTest,TestArtifactCollectorsInteractions,OSXQuarantineTest,CollectorTest,CollectionFilesExportPluginTest,StandardHuntTest,UserTests,TestSearchFileContentWithFixture,TestCollections,ArtifactParserTests,DarwinPersistenceMechanismsParserTest,RDFValueCollectionApiObjectRendererTest,WebHistoryFlowTest,HTTPDataStoreTest,EmbeddedRDFValueTest,DataAgnosticExportConverterTest,GRRSeleniumTest,SqliteDataStoreBenchmarks,TestSystemRootSystemDriveFallbackFlow,OutputpluginsTest,ConfigFileTest,FlowTest,RegistryFlowTest,ExportTest,MatchMethodTests,ResultSetTest,FakeDataStoreTest,HintDefinitionTests,TestAdministrativeFlows,GRRTempFileTestFilename,FilestoreStatsCronFlowTest,ForemanTests,ReportsTest,GRRBaseTest,TestEndToEndTestFlow,DictTest,TestArtifactCollectorsRealArtifacts,KnowledgeBaseUserMergeTest,HandlerTests,NetstatTest,BasicContextTests,ArtifactHandlingTest,ObjcTest,TestWindowsMemory,WindowsActionTests,GrepTest,SystemCronFlowTest,UtilsTest,DataStoreCSVBenchmarks,CheckLoaderTests,TDBDataStoreCSVBenchmarks,ArtifactKBTest,MultiShardedQueueManagerTest,ConfigActionTest,SearchTest,RegistryVFSTests,OSXDriverTests,RekallTestSuite,GeneralFlowsTest,FlowStateTest,TestCryptoTypeInfos,AFF4Benchmark,GRRArtifactTest,GRRFuseDatastoreOnlyTest,TestOSXFileParsing,AFF4SymlinkTest,FlowTestsBaseclass,ConditionTest,RekallVADParserTest
```

Comment:
> No. The purpose of travis is to let you know when tests are failing not to render a nice green button. If some tests pass and some fail then this is how it is - no point turning tests off just to get that nice green feeling :).

The purpose of travis is to let you know if the requested pull is breaking the build or not, and if it's safe to merge. If it is always red, it doesn't serve any purposes, because we're going to be ignoring it. If we adopt a policy of merging only green builds, we'll have to freeze all development until the test suite becomes green as a whole. This will take time and is inconvenient, as other things are going on currently.

I'm not turning tests off to get the nice green feeling. I'm turning tests off so that the rest of the tests, that are enabled, constitute a meaningful build which every new PR can be checked against. Once that is green, we can adopt a policy of merging only green pull requests and requesting from contributors to fix their build if they submit something that is red.

This is exactly the reason why I'll be turning on tests gradually as I fix them. Keep in mind that the failing tests are not failing because the tests are wrong or the implementation is wrong, but because the environment is alien and they need to be migrated carefully. I'll be happy to make an issue for each one and try to fix it individually in subsequent pull requests.

> We dont have to require all new PRs to be travis clean if that is not the case - turning off tests doesnt give you any information anyway since it hides the fact they are failing anyway.

It gives you the information that the tests which are enabled are still succeeding after a pull request will be merged. This gives you confidence that everything that you knew was working previously is still working. 10% coverage is still better than no coverage, so this is meaningful and informative.

If we don't require PRs to be green, then travis doesn't do us any favour. We might as well run tests manually. The automation of the build and the information provided to external contributors that they've done something wrong is what's valuable. If it's always red, there's no point in it, and that's what you're suggesting here.

This technique [has been used successfully in projects such as OpenBazaar](https://github.com/OpenBazaar/OpenBazaar/pull/1106/files) with great results in gradually improved testability for both unit tests and coding style.

> The C++ client will likely require its own test program to run as part of the travis setup and so does not really depend on these tests passing anyway.

It depends on the .travis.yml file being merged.